source/d2sqlite3/database.d

```diff
@@ -1337,12 +1338,14 @@ class SqliteException : Exception
 package(d2sqlite3):
     this(string msg, int code, string sql = null,
          string file = __FILE__, size_t line = __LINE__, Throwable next = null)
+        @safe pure nothrow
     {
-        this("error %d: %s".format(code, msg), sql, code, file, line, next);
+        this(text("error ", code, ": ", msg), sql, code, file, line, next);
```

Comment:
Wait wait. `format` is not nothrow, but `text` is? What black magic is this?

Reply:
`format` can throw due to mismatch between format string and argument. E.g. `format("%d", "Oops this can't be detected at CT");`. Yep.

Reply:
It's so ironic that Walter added `printf` compile-time format string checks, but we don't have it for `format`..

Reply:
I think that `format` is looser in its requirements so it would trigger false positives...

Reply:
And obviously there's no support for D types